File: Creational/Singleton/singleton.py
class Singleton:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super().__new__(cls)
        return cls._instance

    def show_message(self):
        print("Hola desde el Singleton en Python!")

# Ejemplo de uso
if __name__ == "__main__":
    s1 = Singleton()
    s2 = Singleton()

    print(s1 is s2)  # True, misma instancia

    s1.show_message()

File: Creational/Singleton/singleton.cs
using System;

public class Singleton
{
    private static Singleton _instance;
    private static readonly object _lock = new object();

    private Singleton() { }
    public static Singleton Instance
    {
        get
        {
            lock (_lock)
            {
                if (_instance == null)
                {
                    _instance = new Singleton();
                }
                return _instance;
            }
        }
    }

    public void ShowMessage()
    {
        Console.WriteLine("Hola desde el Singleton en C#!");
    }
}

// Ejemplo de uso
class Program
{
    static void Main(string[] args)
    {
        Singleton s1 = Singleton.Instance;
        Singleton s2 = Singleton.Instance;

        Console.WriteLine(s1 == s2);  // True, misma instancia

        s1.ShowMessage();
    }
}